### PR TITLE
setup Repository

### PR DESCRIPTION
**Pull request by phuongha47**
## Related Tickets:
- [#45980](https://edu-redmine.sun-asterisk.vn/issues/45980)

## WHAT this PR does?
- Setup Repositories

## Notes Impacted Areas:
- N/A.

## Attachments:
- N/A.

### CODE REVIEW

**Comment by Khoacannotcode** [comment]
LGTM, great job.

**Comment by phuongha47** [comment]
ready

**Comment by phuongha47** [comment]
ready

**Comment by phuongha47** [comment]
ready

**Comment by phuongha47** [comment]
ready

File: app/Repositories/BaseRepository.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    {
+        return $this->model->create($attributes);
+    }
+
+    public function update($id, $attributes = [])
+    {
+        $result = $this->find($id);
+        if ($result) {
+            $result->update($attributes);
+            return $result;
```
Comment (ghost) [inline app/Repositories/BaseRepository.php]:
```suggestion
            $result->update($attributes);

            return $result;
```

**Review by ghost — APPROVED** [review]

**Pull request merged**